src/Workspaces/Core/Portable/EmbeddedLanguages/VirtualChars/VirtualCharSequence.cs

```diff
@@ -0,0 +1,147 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Immutable;
+using System.Diagnostics;
+using Microsoft.CodeAnalysis.PooledObjects;
+using Microsoft.CodeAnalysis.Text;
+
+namespace Microsoft.CodeAnalysis.EmbeddedLanguages.VirtualChars
+{
+    /// <summary>
```

Comment:
>summary [](start = 9, length = 7)

It seems to me that the purpose of VirtualCharSequence is to expose a portion of a Chunk. 
There seems to be some redundancy which isn't clear: a TextChunk is already capable of doing that...

Would it be simpler to have the following design: `abstract class VirtualCharSequence` with two children `ImmutableArrayVirtualCharSequence` and `Text...`, and removing the `Chunk` types?

Reply:
@jcouv easier to explain on gitter.  want to hop over?  the main problem is immutability and allocations.  If that's not a clear enough lede, i'm happy to go over more over there!